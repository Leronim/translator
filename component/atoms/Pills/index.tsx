import React from 'react';
import { View, StyleSheet, Text } from 'react-native';

interface ListItemProps {
    text: string
    id: number
    index: number
}


const Pills: React.FC<ListItemProps> = ({ text, id, index}: ListItemProps) => {
    return(
        <View style={styles.root}>
            <View style={styles.shadow}>
                <Text>{text}</Text>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    root: {
        marginRight: 10,
        marginTop: 15,
        width: 70,
        height: 30,
    },
    shadow: {
        shadowOffset: {
            width: 0, 
            height: 4
        },
        justifyContent: 'center',
        alignItems: 'center',
        shadowOpacity: 0.25,
        shadowColor: "#000",
        shadowRadius: 1,
        borderRadius: 13,
        backgroundColor: 'white',
        width: 70,
        height: 30,
        borderWidth: 1,
        borderColor: '#C9C9C9',
    },
})

export default Pills