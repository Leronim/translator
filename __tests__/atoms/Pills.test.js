import React from 'react';
import { Text } from 'react-native';
import Pills from '../../component/atoms/Pills';
import { render } from '@testing-library/react-native';
test('Pills render', () => {
    const tree = render(<Pills/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('Pills message', () => {
    const message = "React";

    const { UNSAFE_getAllByType } = render(<Pills text={message}/>);

    const [ text ] = UNSAFE_getAllByType(Text)

    expect(text.props.children).toEqual(message);
})