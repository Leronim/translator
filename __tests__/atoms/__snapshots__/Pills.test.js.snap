// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pills render 1`] = `
<View
  style={
    Object {
      "height": 30,
      "marginRight": 10,
      "marginTop": 15,
      "width": 70,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#C9C9C9",
        "borderRadius": 13,
        "borderWidth": 1,
        "height": 30,
        "justifyContent": "center",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 1,
        "width": 70,
      }
    }
  >
    <Text />
  </View>
</View>
`;
